Added filename alias field in settings & export menus

mihail-alexe-nutanix opened the pull request:
Added `Filename` fields in the `Settings` and `URL Export` screens, next to the existing one in `Export to file`.

The reason for adding it is more to accommodate an use case that we're currently having in Nutanix when exporting to URL, where UX wants to have friendly names on the Figma files and the ability to (re)name these files as they please, while engineering need a bit of a more standardized file naming pattern, especially when automating with GH Actions.

As an example, see the table below

| Figma file name                       | Alias name (sent to GH Actions) |
|---------------------------------------|---------------------------------|
| Core Design Tokens - Light            | light.core.tokens.json          |
| Core Design Tokens - Dark             | dark.core.tokens.json           |
| Core Design Tokens - Theme X          | theme-x.core.tokens.json        |
| Kittens team super awesome UX [Light] | light.kittens-team.tokens.json  |
| ...                                   | ...                             |

Please let me know if I need to do any further improvements and/or add test in order to accept this contribution.

lukasoppermann commented:
Hey @mihail-alexe-nutanix,

sorry I just noticed the PR now.

I will have a look and provide some more feedback.

But I already have one change. Can you please remove changes to `package.json` and `.lock`. Also please reset `src/utilities/version.ts` adn remove the `dist` files. This will all be done by the script when publishing.

I don't quick get why there is a need for it in settings. My assumption was that we don't need it for github as you have the filename property but I understand your issue.

However why do we need it in `settings`?

mihail-alexe-nutanix commented:
@lukasoppermann I have reverted the changes that you've requested. Sorry for that, but I couldn't find anything about versioning and releasing, so I assumed it's a manual process.

As for `settings`, we don't really need it in there. I have added it there for simplicity, to set it together with all the other options when changing settings. But it is indeed a bit redundant.
Just let me know what you think + other feedback you've got, and I will take care of it.

lukasoppermann commented:
Okay, there was a missunderstanding on my side. 
You are using the same filename for both export file and send to server. In this case I would prefer to only add it the the main settings screen (this has all the common settings).

Does that make sense? Looking forward to you updates.

lukasoppermann commented:
Okay, looks good, there are a few more comments and please to a rebase (I added some of your improvements already to main to clean this PR up, as they have nothing to do with the actual PR and just create noise in here, but are valuable).

Thanks

lukasoppermann commented:
Hey @mihail-alexe-nutanix it looks really good. If you could fix the few comments that are still open, I will be able to merge it.

lukasoppermann commented:
hey @mihail-alexe-nutanix does it work now?

mihail-alexe-nutanix commented:
Hey @lukasoppermann!
Yes, I can see them now. I'll take care of them ASAP. Thank you!

mihail-alexe-nutanix commented on src/ui/components/UrlExportSettings.tsx:
```diff
@@ -145,16 +146,17 @@ export const UrlExportSettings = () => {
       <Footer>
         <WebLink align='start' href='https://github.com/lukasoppermann/design-tokens#design-tokens'>Documentation</WebLink>
         <CancelButton />
-        <Button type='button' onClick={handleFormSubmit} autofocus>Save & Export</Button>
```
I have tried this one out with an invalid token, invalid URL, token with not enough privileges and while offline (wifi turned off), comparing the released plugin (production version) and the my version. I couldn't find any problem... In all the listed situations I got toast errors with the appropriate message. Could you please provide some reproduction steps for the issues you've found?

I would like to focus on fixing the "no error" bug, rather than reverting this, mostly for UX purposes. The arguments are the following:

- In the current (production) implementation, although there's an `onSubmit` handler on the form, there is no way to submit on `enter` in an input field, because of some react quirks. It strictly needs a submit button within the form to call handler.

- The other problem is that when clicking the button, `handleFormSubmit` receives as `event.target` the actual button element. This causes `checkValidity()` to return `true`, even when the form is invalid (because buttons, as input elements, are always valid). This means the action is attempted (save/export) and not prevented and errors highlighted.

  - For example submitting without a GH Actions URL (or any other required input), instead of highlighting the error and allowing me to input the value, it closes the popup and displays a toast error with `🚨 No server url was provided, push aborted!`

lukasoppermann replied:
Oh, wow, thanks. I will check this out and let you know what I find. I'd love to get this fixed.

lukasoppermann replied:
Okay, so if I add the `required` and change to `submit` removing the `onClick={handleFormSubmit}`

If I set the url to `https://test.com` and click the button or press return, I get the blank screen issue. Any idea?

mihail-alexe-nutanix replied:
Just add `event.preventDefault()` in the `handleSubmit` function.
Otherwise the form will make the Figma browser to navigate, and since it's not a real web app, will display a blank screen.
